packages/table/src/table-column.js

```diff
@@ -444,6 +449,10 @@ export default {
       columnIndex = [].indexOf.call(parent.$el.children, this.$el);
     }
 
+    if (this.$scopedSlots.header) {
```

Comment:
For selection column, header scope-slot is not allowed to set. When users use this slot, they should get warning information.